Sources/reactant-ui/GenerateCommand.swift

```diff
@@ -192,13 +218,119 @@ class GenerateCommand: Command {
+                ApplicationTheme.selector.register(target: ReactantLiveUIManager.shared, listener: { theme in
+                    ReactantLiveUIManager.shared.setSelectedTheme(name: theme.name)
+                })
+            """
+
+            output.append(ifSimulator(swiftVersion: swiftVersion, commands:liveUIActivation))
         }
         output.append("}")
 
         try output.joined(separator: "\n").write(to: outputPathURL, atomically: true, encoding: .utf8)
     }
 
+    private func theme(context: GlobalContext, swiftVersion: SwiftVersion) throws -> String {
+        let description = context.applicationDescription
+        func allCases<T>(item: String, from container: ThemeContainer<T>) throws -> String {
+            return try description.themes.map { theme in
+                guard let themedItem = container[theme: theme, item: item] else {
+                    throw GenerateCommandError.themedItemNotFound(theme: theme, item: item)
+                }
+                let typeContext = SupportedPropertyTypeContext(parentContext: context, value: themedItem)
+                return "case .\(theme): return \(themedItem.generate(context: typeContext))"
+            }.joined(separator: "\n")
+        }
+
+        let cases = description.themes.map {
+            "    case \($0)"
+        }.joined(separator: "\n")
+
+        let allColorsSorted = try description.colors.allItemNames.sorted().map { """
+            public var \($0): UIColor {
+                switch theme {
+                \(try allCases(item: $0, from: description.colors))
+                }
+            }
+            """
+        }
+        let allImagesSorted = try description.images.allItemNames.sorted().map { """
+            public var \($0): UIImage? {
+                switch theme {
+                \(try allCases(item: $0, from: description.images))
+                }
+            }
+            """
+        }
+        let allFontsSorted = try description.fonts.allItemNames.sorted().map { """
+            public var \($0): UIFont {
+                switch theme {
+                \(try allCases(item: $0, from: description.fonts))
+                }
+            }
+            """
+        }
+
+        let rxSwiftShim: String
+        // canImport is only available from 4.1 and above
+        if swiftVersion >= .swift4_1 {
+            rxSwiftShim = """
+            #if canImport(RxSwift)
+            import RxSwift
+
+            extension ReactantThemeSelector.ListenerToken: Disposable {
+                public func dispose() {
+                    cancel()
+                }
+            }
+            #endif
+            """
+        } else {
+            rxSwiftShim = ""
+        }
+
+        return """
+
+        public enum ApplicationTheme: String, ReactantThemeDefinition {
+            public static let selector = ReactantThemeSelector<ApplicationTheme>(defaultTheme: .\(description.defaultTheme))
+
+        \(cases)
```

Comment:
I assume the texts themselves are indented instead. Is that wise?

Reply:
Well. Let's not look at the generated swift code. This is something we should solve as a separate PR.

Reply:
Why is that?

Reply:
This is not inside the `UIGenerator`, but in `GenerateCommand` where it uses simple `var output: [String]` and `output.append(...)`. Because of that, there's not even basic auto-indentation. The indentation in `UIGenerator` is done by the `l` function.